docs/compilation.rst

```diff
@@ -1,50 +1,78 @@
 
 Python binding
-=================
+==============
 
-We use `pybind11 <https://github.com/pybind/pybind11>`_ to build the Python binding. It tries to automatically detect the installed version of Python and link against that. When this fails, or when there are multiple versions of Python and it finds the wrong one, delete CMakeCache.txt and then invoke CMake as follows:
+We use `pybind11 <https://github.com/pybind/pybind11>`_ to build the Python binding.
+It tries to automatically detect the installed version of Python and link against that.
+When this fails, or when there are multiple versions of Python and it finds the wrong one, delete CMakeCache.txt and then invoke CMake as follows:
 
 .. code-block:: bash
 
     cmake -DPYTHON_EXECUTABLE:FILEPATH=<path-to-python-executable> ../src
 
 .. Note:: Python binding issues can also refer to `pybind11 document page <http://pybind11.readthedocs.io/en/stable/faq.html>`_.
 
-If you do not want Python binding, turn off the compilation options ``Open3D_BUILD_PYTHON_BINDING`` and ``Open3D_BUILD_PYTHON_BINDING_TESTS``.
+If you do not want Python binding, you may turn off the following compilation options:
+
+- ``BUILD_PYBIND11``
+- ``BUILD_PYTHON_MODULE``
+- ``BUILD_PYTHON_TESTS``
+- ``BUILD_PYTHON_TUTORIALS``
 
 Dependencies
-================
+============
+
+Open3D dependencies are included in ``src/External`` folder.
+The user has the option to force building the dependencies from source or to let CMake search for installed packages.
+If a build option is turned OFF and CMake can't find its corresponding package the configuration step will fail.
 
-On Windows, all dependent libraries are built from source code. Dependencies are included in ``src/External`` folder.
+Example error message:
 
-On Linux and OS X, the default behavior of Open3D is to find packages installed on the OS and link to them. If the required package(s) is not found, it is built from source code. This avoids compilation of dependent libraries as much as possible, and thus significantly reduces compilation time. However, sometimes there can be compatibility issues due to version conflict. In this case, you can manual turn off the compilation option ``Open3D_USE_NATIVE_DEPENDENCY_BUILD`` to force building dependent libraries from source code:
+| ``CMake Error at External/CMakeLists.txt:32 (message):``
+| ``EIGEN3 dependency not met.``
+
+The following is an example of how to force building from source a number of dependencies:
 
 .. code-block:: bash
 
-    cmake -DOpen3D_USE_NATIVE_DEPENDENCY_BUILD=OFF ../src
+    cmake -DBUILD_EIGEN3=ON  \
+          -DBUILD_GLEW=ON    \
+          -DBUILD_GLFW=ON    \
+          -DBUILD_JPEG=ON    \
+          -DBUILD_JSONCPP=ON \
+          -DBUILD_PNG=ON     \
+          ../src
+
+.. tip:: This can save a lot of time on Windows where it can be particularly difficult to install the Open3D dependencies.
+
+.. note:: Enabling these build options may increase the compilation time substantially.
```

Comment:
I don't think it will substantially increase the time :) Let's just say 'increase the compilation time'

Reply:
got it.